### PR TITLE
Added Admin page and integration to store

### LINKED ISSUE #7: Create an admin page
## Acceptance criteria
* Adds the teacher overview component
* Adds the student overview component

Dependant on #33

### PR DESCRIPTION
**Pull request by te4-adrian-almetun-smeds**
## What issue did you implement or fix?
Fixed #7

## Summary
- Added admin page
- Dynamically fetches data from the store

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## How can this implementation be improved?
- Change bar into zar. :shipit:

### CODE REVIEW

File: lib/src/components/UserOverview.vue [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  //   return {
+  //     items: [
+  //       {
+  //         email: "hehehe",
+  //         name: "hohoho",
+  //         teacher: "Daniel Berg"
+  //       }
+  //     ]
+  //   };
+  // },
```
Comment (te4-adrian-almetun-smeds) [inline lib/src/components/UserOverview.vue:54]:
```suggestion
```

This probably should be removed

File: lib/src/components/UserOverview.vue [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+  computed: {
+    teachers: function() {
+      return store.getters["users/teachers"].map((x: any) => x.name);
+    },
+    items: function() {
```
Comment (te4-adrian-almetun-smeds) [inline lib/src/components/UserOverview.vue:60]: Can we rename this variable/function to something more specific?
I would suggest `users`

File: lib/src/components/UserOverview.vue [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      return store.getters[`users/${this.type}s`];
+    }
+  },
+
+  methods: {
+    getNameFromID: function(id) {
```
Comment (te4-adrian-almetun-smeds) [inline lib/src/components/UserOverview.vue:66]:
```suggestion
    getTeacherNameFromID: function(id) {
```
This change would require all references to be updated as well.

File: lib/src/store/modules/users.ts [hunk 4]
```diff
@@ -82,9 +83,10 @@ export default {
       return state.users;
     },
     teachers: (state: any) => {
-      return state.users.filter((x: User) => {
-        x.type === "teacher";
-      });
+      return state.users.filter((x: User) => x.type == "teacher");
+    },
+    students: (state: any) => {
```
[the remark this answers is not in the document]
Comment (te4-adrian-almetun-smeds) [reply lib/src/store/modules/users.ts:88]: As mentioned in #65 fixing this is something we are not going to put a effort into at this current time.

File: lib/src/views/Admin.vue [hunk 5]
```diff
@@ -0,0 +1,39 @@
+<template>
+  <v-container>
+    <UserOverview type="teacher" :headers="teacherHeaders" class="mb-5" />
```
Comment (te4-adrian-almetun-smeds) [inline lib/src/views/Admin.vue:3]: What does this (`mb-5`) class do?
As it's only used for the first component. Is that intentional?

**Review by te4-henrik-stahl — APPROVED** [review]
wow i like

**Pull request merged**